### ISSUE TITLE
Main delay calculation workflow: Look Direction is unused

### ISSUE THREAD

**Issue by garlic-os** [dbekaert/RAiDER#676]
The `look_dir` parameter as provided in a YAML file is never used. It is accessed in one spot to validate that it is either "left" or "right", and never again. A vscode all files search for the term "look_dir" only produces results from inside other data structures and that one validation check, and none where the `look_dir` from a YAML run config gets used.

Do we not use Look Direction anymore, or is it being left out somewhere on accident?

**Desktop**
 - RAiDER git tag:
commit 2a348023801fc1f7fb21427d3ae63150530e2382 (HEAD -> dev, tag: v0.5.2, upstream/main, upstream/dev, upstream/HEAD)
Merge: 223d862 2b7190c
Author: Jeremy Maurer <maurer.jeremy@gmail.com>
Date:   Mon Jul 15 11:54:29 2024 -0500
&nbsp;&nbsp;&nbsp;&nbsp;Merge pull request #670 from dbekaert/dev
&nbsp;&nbsp;&nbsp;&nbsp;v0.5.2
 - OS: Arch Linux 5.15.153.1-microsoft-standard-WSL2 on Windows 10

**Comment by dbekaert** [comment]
Not all sensors have same look direction. Will defer to @jlmaurer if this was a placeholder for the future. E.g. NISAR/ CSK etc.

**Comment by jlmaurer** [comment]
@dbekaert @garlic-os yes this will depend on the sensor. Right now everything is right-looking, but very soon we will have a left-looking sensor (NISAR). We may want to take a look at the workflow and make sure the look direction is properly carried through.

**Comment by s-sasaki-earthsea-wizard** [comment]
@jlmaurer's 2024 reply here was that this is latent — *"Right now everything is right-looking, but very soon we will have a left-looking sensor (NISAR)"*. That sensor is now flying, so I checked #676 against the data. Two findings, the second stronger than I expected:

1. The gap is still open on `dev` (1fb9e14): the run config's `look_dir` is structurally unable to reach `Raytracing`.
2. **NISAR is not "sometimes left-looking" — it is left-looking, full stop.** All 221 (track, pass) groups across the 10,217-granule `NISAR_L2_GUNW_BETA_V1` collection are `Left`, as are sampled GSLC/GCOV/RSLC products. There is no right-looking NISAR acquisition. So the hardcoded `right` is wrong for **every** NISAR product, not just some.

Evidence, reports, and a reproducible harness:
**https://github.com/s-sasaki-earthsea-wizard/raider-nisar-lookdir/tree/58260af9f7c071ce6f3b807262fd8ffa9d940d35**

- [`reports/lookdir_verification.md`](https://github.com/s-sasaki-earthsea-wizard/raider-nisar-lookdir/blob/58260af9f7c071ce6f3b807262fd8ffa9d940d35/reports/lookdir_verification.md) — depth: 10 granules, metadata vs. re-derived geometry, buffer consequence.
- [`reports/archive_survey.md`](https://github.com/s-sasaki-earthsea-wizard/raider-nisar-lookdir/blob/58260af9f7c071ce6f3b807262fd8ffa9d940d35/reports/archive_survey.md) — breadth: the archive-wide census.

It reads NISAR metadata over HTTP range requests, so `make setup && make verify` reproduces the first report in ~2 min / ~380 MB instead of downloading ~18 GB, and `make survey` redoes the whole census in ~6 min.

## 1. The gap is still present on `dev` (1fb9e14)

`look_dir` is validated at `cli/raider.py:105` and stored as `RunConfig.look_dir` at `cli/raider.py:128` — and that's the end of the line; nothing else reads it. `Raytracing` is constructed in exactly two places, and neither forwards it:

```python
# tools/RAiDER/cli/validators.py:54-67
def get_los(los_group: LOSGroupUnparsed) -> LOS:
    if los_group.orbit_file is not None:
        if los_group.ray_trace:
            los = Raytracing(los_group.orbit_file)          # look_dir defaults to 'right'
    elif los_group.los_file is not None:
        if los_group.ray_trace:
            los = Raytracing(los_group.los_file, los_group.los_convention)   # ditto
```

`get_los()` only receives `LOSGroupUnparsed`, which has no `look_dir` field — `look_dir` lives at the top level of the run config, so it can't reach `Raytracing.__init__` even in principle. Setting `look_dir: left` in a YAML today is silently a no-op.

## 2. Every NISAR product states its own look direction — and says `Left`

All NISAR products carry `science/LSAR/identification/lookDirection` (`|S4`/`|S5`, `description = 'Look direction, either "Left" or "Right"'`), so RAiDER never has to guess.

Census of `NISAR_L2_GUNW_BETA_V1` (10,217 granules → 221 distinct (track, pass) groups, one probed per group):

| `lookDirection` | groups |
|---|---|
| **`Left`** | **221 / 221** |

Sampling the other beta collections across distinct tracks gives `Left` for GSLC (5/5), GCOV (5/5) and RSLC (5/5).

I didn't take the field on faith. For 10 granules inspected in depth — the JPL sample-suite product, the 7-granule track-159 stack over central Spain, plus track 172 ascending and track 064 descending — the look side was independently re-derived from the state vectors at `science/LSAR/GUNW/metadata/orbit/reference/{time,position,velocity}` and the `identification/boundingPolygon` footprint, by solving the zero-Doppler condition and testing the sign of `(T−P) · (V × P)`. Metadata and geometry agree **10/10**.

The one `Right` in that set is the JPL sample-suite granule, which is ALOS-1 PALSAR surrogate data rather than a NISAR acquisition. I kept it deliberately as a control: it shows the reader and the geometric check discriminate both values instead of always reporting `Left`.

## 3. Concrete consequence today

`cli/raider.py:264` buffers the weather model with `los.getLookDirection()`, which for a ray-traced run is always the hardcoded `Right`. Running `llreader.AOI.calc_buffer_ray` on the real footprints puts the buffer on the wrong side for all 9 NISAR granules — e.g. `..._003_159_A_022_...` (S,N,W,E = 39.37, 41.885, −4.519, −0.944; ascending; `Left`):

```
lookDir=right (RAiDER's actual value) -> W,E = -5.14, -0.94   # buffered WEST
lookDir=left  (what the product says) -> W,E = -4.52, -0.32   # buffered EAST
```

The weather model gets extended ~0.62° *away from* the sensor rather than toward it. The same wrong `LookSide` is then handed to `isce3.geometry.geo2rdr` in `Raytracing.getLookVectors`.

## 4. Two things that seem worth separating

**(a) Plumb the existing parameter.** The minimal fix is to get `look_dir` into `get_los()` and on to `Raytracing(...)` — either passing it as an argument to `get_los`, or moving `look_dir` into the `los_group`. Small and self-contained; happy to open a PR. Any preference on which shape you'd want?

**(b) For NISAR, prefer the product over the run config.** Given the census above, a YAML `look_dir` that defaults to `right` is wrong-by-default for every NISAR scene, and asking users to hand-set it is a footgun with no upside — the GUNW already states the answer. Reading `identification/lookDirection` seems strictly better. Note the ARIA-side helper won't generalize:

```python
# tools/RAiDER/aria/prepFromGUNW.py:249
def get_look_dir(self) -> LookDir:
    look_dir = self.path_gunw.name.split('-')[3].lower()   # 'S1-GUNW-A-R-064-...' -> 'R'
    return 'right' if look_dir == 'r' else 'left'
```

NISAR granule IDs are underscore-delimited and carry no look field (`NISAR_L2_PR_GUNW_003_159_A_022_...` — the `A` is the pass direction), so this filename parse returns `left` for *every* NISAR product. It would happen to be right for the wrong reason on real data, and wrong on the right-looking sample product above.

Is a NISAR ingest path (analogous to `aria/prepFromGUNW.py`) already planned or underway? I don't want to duplicate work — but if (a) is useful as a standalone fix, I can put that up now.

---

*Disclosure: these investigations were written with AI assistance — Claude Opus 4.8 and Fable 5 (Anthropic), and GPT‑5.6 Terra (OpenAI). All code was reviewed by the author and reproduced the reported results before posting.*
